man/tools.go

```diff
@@ -0,0 +1,7 @@
+//go:build tools
+
+package man
+
+import (
+	_ "github.com/cpuguy83/go-md2man/v2"
+)
```

Comment:
We should have a look (in the CLI as well) if we can use the updated `cli-docs-tools` functionality to generate both the man-pages and the markdown variant for the daemon; in that case we may no longer need the separate `md2man` binary to produce the man-pages;

- https://github.com/docker/cli-docs-tool/pull/14

I think we should move _both_ the man-page _and_ the `dockerd.md` documentation back into this repository; the `dockerd.md` markdown documentation is currently 100% manually maintained, but with the `cli-docs-tool` can generate the "options" table with flag descriptions, combining it with manually maintained content from the markdown document; similar to https://github.com/docker/cli/blob/v27.1.1/docs/reference/commandline/container_run.md

Reply:
I gave it a shot and the results were not pretty. The auto-generated option descriptions are much more terse than those in the hand-written dockerd man page, which contains some multi-paragraph option descriptions that would be inappropriate for `dockerd --help` to output. And the `ENVIRONMENT VARIABLES` table that cli-docs-tools pulls in from dockerd.md when generating the man page is nearly unreadable on a terminal.

I don't think we should try to to generate the dockerd man page from Go sources at this time. It is too lossy.

Moving dockerd.md into this repository so that the options table can be kept up to date with auto-generated descriptions sounds reasonable. But it's orthogonal to this work. Could we do that work as a follow up to avoid boiling the ocean?